exchanges/bitstamp/bitstamp_websocket.go

```diff
@@ -101,137 +117,190 @@ func (b *Bitstamp) wsHandleData(respRaw []byte) error {
 			}
 		}() // Connection monitor will reconnect
 	case "data":
-		wsOrderBookTemp := websocketOrderBookResponse{}
-		err := json.Unmarshal(respRaw, &wsOrderBookTemp)
-		if err != nil {
+		if err := b.handleWSOrderbook(context.TODO(), wsResponse, respRaw); err != nil {
 			return err
 		}
-		var currencyPair string
-		splitter := strings.Split(wsResponse.Channel, currency.UnderscoreDelimiter)
-		if len(splitter) == 3 {
-			currencyPair = splitter[2]
-		} else {
-			return errWSPairParsingError
-		}
-		pFmt, err := b.GetPairFormat(asset.Spot, true)
-		if err != nil {
+	case "trade":
+		if err := b.handleWSTrade(context.TODO(), wsResponse, respRaw); err != nil {
 			return err
 		}
-
-		enabledPairs, err := b.GetEnabledPairs(asset.Spot)
-		if err != nil {
+	case "order_created", "order_deleted", "order_changed":
+		if err := b.handleWSOrder(context.TODO(), wsResponse, respRaw); err != nil {
 			return err
 		}
+	default:
+		b.Websocket.DataHandler <- stream.UnhandledMessageWarning{Message: b.Name + stream.UnhandledMessage + string(respRaw)}
+	}
+	return nil
+}
 
-		p, err := currency.NewPairFromFormattedPairs(currencyPair, enabledPairs, pFmt)
-		if err != nil {
-			return err
-		}
+func (b *Bitstamp) handleWSOrderbook(_ context.Context, wsResp *websocketResponse, msg []byte) error {
```

Comment:
rm param `_ context.Context,`

Reply:
Yeah, okay.
When I finally crusade for universal context it won't make any difference if a few places already had it, especially if they were ignoring it.
Fixed: 5aedf4ad